Cookbook/Technical-Documents/Architecture.md

```diff
@@ -0,0 +1,309 @@
+Babylon iOS Architecture
+========================
+
+## Overview
+
+In Babylon, we use [Model-View-ViewModel (MVVM)](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93viewmodel) architecture pattern to develop each GUI screen and its business logic.
+The most common MVVM practice in iOS is to treat `UIViewController` as `View`, which holds `ViewModel` to delegate the event and business logic handling to decouple from `View`'s primary goal of "rendering its appearance".
+
+While MVVM architecture gained its popularity in iOS community as the rise of Functional Reactive Programming (FRP, e.g. [ReactiveSwift](https://github.com/ReactiveCocoa/ReactiveSwift)) with its easy data-binding mechanism, there are still concerns about the following issues:
+
+1. How to manage states (in more testable way)
+2. How to deliver events (in simpler way than data-bindings)
+3. How to setup MVVM (single screen)
+4. How to coordinate / scale multiple MVVMs (mutliple screens with navigation / single screen with nested MVVMs)
+
+Unfortunately, FRP itself does not offer a best practice for these questions, since it's main purpose is to build small pipelines through reactive operator compositions, splitting and combining their stateful representation, to make a large application.
+It's a bottom-up development tool, so states and event-flows can easily get scattered and spaghetti-ed across the entire codebase if we don't share the same architectural mindset among the team.
+
+Luckily, web-development community has reached a more consistent architectural solution to use [Redux](https://redux.js.org) as a predictable (global) state container alongside [React](https://reactjs.org/), a state (or virtual-view) diffing renderer.
+Redux encourages the developers to apply uni-directional flow architecture (inspired by [Elm](https://elm-lang.org/)) and centralize state management (powered by [Automata Theory](https://en.wikipedia.org/wiki/Automata_theory)) to solve our questions 1 and 2.
+Moreover, React brings a new UI paradigm that can be rendered just from a state via Redux.
+
+So, in Babylon iOS team, we created [ReactiveFeedback](https://github.com/Babylonpartners/ReactiveFeedback) and [Bento](https://github.com/Babylonpartners/Bento) as their counterparts to solve 1 and 2 that nicely fits on top of existing UIKit framework.
+
+Also, for 3 and 4 (navigation), we defined `Builder` and `FlowController` layers to satisfy the remaining missing pieces.
+
+So, in a nutshell:
+
+1. `ViewModel` (with ReactiveFeedback)
+2. `Renderer` (Bento Virtual-View)
+3. `Builder`
+4. `FlowController`
+
+are the 4 core entities to achieve Babylon iOS Architecture, as well as:
+
+- `Bento.BoxViewController` (universal class for rendering `UITableView`)
```

Comment:
I think `BoxViewController` is the most commonly used ViewController class to explain about our architecture.
Though we might be tempted to introduce its variety, it is not very important in this documentation since this topic is about brief overview for how each layer interacts rather than in-depth discussion of each layer.